src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -155,7 +157,10 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
         std::scoped_lock<std::shared_mutex> lock(m_mutex);
         if (parsedMessage.at("type") == "offsets")
         {
+            // File processing started.
+            m_fileProcessInProgress.store(true);
             auto jsonPointer {"/data"_json_pointer};
+
```

Comment:
Severe (design): raw type processing is not being protected

We can secure both sections with this:
```suggestion
        // Lock the mutex to protect the access to the internal databases.
        std::scoped_lock<std::shared_mutex> lock(m_mutex);
        m_fileProcessInProgress.store(true);
        DEFER(
            [&]()
            {
                m_fileProcessInProgress.store(false);
                m_contentRegistration->wakeUpThread();
            });

        if (parsedMessage.at("type") == "offsets")
        {
            // File processing started.
            auto jsonPointer {"/data"_json_pointer};
```

The DEFER ensures that the lambda described is run when we go out of scope

> [!NOTE]
> You will need to add the header #include "defer.hpp"